File: python/001/twoSum.py
class Solution:
    def twoSum(self, nums, target: int):
        hashdict = {}
        for i, num in enumerate(nums):
            complement = target - num
            if complement in hashdict and i != hashdict[complement]:
                return [i, hashdict[complement]]
            hashdict[num] = i
        
        return hashdict

File: php/001/twoSum.php
class Solution {

    /**
    * @param Integer[] $nums
    * @param Integer $target
    * @return Integer[]
    */
    function twoSum($nums, $target) {
        $temp = $nums;
        $result = [];
        foreach ($nums as $key => $num) {
            $res = $target - $num;
            unset($temp[$key]);
            if (in_array($res,$temp)) {
                foreach ($nums as $k => $n) {
                    if ($n == $res) {
                        $result = [$key,$k];
                    }
                }
            }
        }
        return $result;
    }
}